### DISCUSSION TITLE
Brave does not recognize the Coapp

### DISCUSSION THREAD

**Discussion by cwsooner** [aclap-dev/video-downloadhelper#854] [Help and Q&A]
### Your question

The terminal shows that the Coapp is installed and ready to use which it is in FF but Brave still tells me that the Coapp is required to download certain videos.  I can't seem to find much help on Brave and the Coapp.  I was able to download a couple of videos from Video Haven which I have not been able to in the past from any site through Brave without the Coapp.  

### Addon & CoApp version and details

VDH version 9.2.3
The browser you're using (be as specific as possible):  Brave [Version 1.66.118 Chromium: 125.0.6422.147 (Official Build) (64-bit)](https://brave.com/latest/)
OS (Linux/Windows/Mac) & OS version:  Ubuntu 22.04 LTS
OS 64bits or 32bits:  64bit
Link to the coapp you installed:  curl -sSLf https://github.com/aclap-dev/vdhcoapp/releases/latest/download/install.sh | bash
On Linux, your sandbox mechanism (if any): flatpak/snap/…Snap but the installation showed flatpak in it I think.  
On Linux, the output of /opt/vdhcoapp/vdhcoapp --info (or ~/.local/share/vdhcoapp/vdhcoapp --info if it was install with the script)  
"id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/home/guardian/.local/share/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/guardian",
  "converterBinary": "/home/guardian/.local/share/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.  In addition, I know diddly squat, perhaps less, about Linux.

Search this forum with "recognize coapp" as your search key.  Make sure your Filter specifies All, not some other value.  Perhaps you will find relevant help there.  I must add that Brave is not one of the browsers supported by VDH.  After Firefox, Chrome, & Edge, you are strictly on your own.

That search may force you to do a lot of reading.  But consider.  Your fellow users posted here for a reason.  That reason is to help you.  If you ignore what they say, you'll be thwarting their efforts to help you.  That would be rude.  You don't want to be rude, do you?  I didn't think so.  So bite the bullet & start reading.

**Reply by cwsooner** [reply]
What are you talking about?  While I appreciate your reply I have been reading other replies on the Coapp not being recognized......by Firefox.  I only use FF when I have to.  It's not a safe go to browser for me anymore.  I don't use Chrome and I hate Microsoft.  That's why I use Ubuntu. 
I have seen your name in many other strings by the way so I have been reading and almost NO ONE mentions using Brave which is mentioned in with the other browsers in one of the dowload helpers pages or somewhere of which I can't seem to locate at the moment.  
Also, Brave is based off of Chrome anyway.

**Reply by GrampaWildWilly** [reply]
Somebody did something with a json file.  He had to copy it from one place to another.  But since it was on Linux & I don't know Linux, the details just fell out of my head.  If you can find that discussion, it might help you.

**Reply by GrampaWildWilly** [reply]
When you search here on GitHub, you should make sure your Filter has All checked & not some other value.

**Comment by Mattmjs** [comment]
You should verify what brave browser app your using. There might be something like about brave in the settings somewhere.
That'll tell you if it is a snap or flatpack. Also the firefox snap on Ubuntu apparently has a popup once the companion app is installed.
This is to give it permission to allow it to connect to the companion app. I wonder if there was similar pop up for brave that you
need to approve.

**Reply by cwsooner** [reply]
You mean like this?  [Version 1.66.118 Chromium: 125.0.6422.147 (Official Build) (64-bit)](https://brave.com/latest/)

**Reply by cwsooner** [reply]
No popup for Brave.  I don't remember one for FF.  I used it to compare and see if it would work or have the same result but it worked without having to do anything I think.  Well, it just wouldn't work on that particular video site, Streamyard but it had a report button that popped up and I sent the report.  I will upload a screenshot of the results I got from Brave.

**Reply by cwsooner** [reply]
![Screenshot from 2024-06-12 17-46-58](https://github.com/aclap-dev/video-downloadhelper/assets/36794121/928e0c07-6c01-482d-9bcd-2d37a3ff327b)

**Reply by cwsooner** [reply]
![Screenshot from 2024-06-12 17-47-18](https://github.com/aclap-dev/video-downloadhelper/assets/36794121/671f2e8c-d662-44cb-8d1e-2e7a67a47796)

**Reply by cwsooner** [reply]
Yet the companion app is already installed.

**Reply by Mattmjs** [reply]
I'd like you to open the settings in the extension, it'll try to check the companion. Does it have the same message as in your image
or a different message. I have two scenarios, it's not finding the native messaging json file if in the wrong location.
Maybe the first point is correct and is in correct location, that would lead the to the second. The browser being in a sandbox has not allowed permission to connect to companion app.

I might even have a look in a virtual machine to see how it behaves.

**Reply by cwsooner** [reply]
![Screenshot from 2024-06-13 15-26-00](https://github.com/aclap-dev/video-downloadhelper/assets/36794121/c54b1307-bc4b-4ce1-a768-6debaffa0bf5)

**Reply by Mattmjs** [reply]
I tried taking a look in a virtual machine, but it wouldn't download the browser. Anyway in your image it doesn't find native
messaging host. Your have to find out if it's in the right location or not. Look at this link below & check the locations
for chromium. https://developer.chrome.com/docs/extensions/develop/concepts/native-messaging#native-messaging-host-location
